src/main/java/com/box/sdk/BoxSearchParameters.java

```diff
@@ -318,7 +317,7 @@ private String printParam(Object obj)    {
      * @param    paramValue Object that will be checked if null.
      * @return this.true if the parameter that is being checked is not null
      */
-    public boolean isNotNull(Object paramValue) {
+    private boolean isNullOrEmpty(Object paramValue) {
```

Comment:
The logic of this method needs to be flipped. Right now it returns true if the object is not null.